keys-component.js

```diff
@@ -69,40 +72,59 @@ function Ctrl($scope, $routeParams, brAlertService, brKeyService,
     self.modals.showRevokeKeyAlert = true;
     self.modals.key = key;
   };
+
   self.confirmRevokeKey = function(err, result) {
     if(!err && result === 'ok') {
-      self.service.revoke(self.modals.key.id);
+      self.service.revoke(self.modals.key.id)
+        .catch(function(err) {
+          brAlertService.add('error', err, {scope: $scope});
+        })
+        .then(function() {
+          setActiveKeys();
+          $scope.$apply();
+        });
     }
     self.modals.key = null;
   };
 
   self.onKeyGenerate = function(key) {
+    self.activeKeys = true;
     return self.service.collection.add(key);
   };
 
+  function setActiveKeys() {
+    self.activeKeys = false;
+    for(var i = 0; i < self.keys.length; i++) {
+      if(self.keys[i].sysStatus === 'active') {
+        self.activeKeys = true;
+        break;
+      }
+    }
+  }
+
```

Comment:
It's not in the dependency list yet (not used anywhere else), so let's just leave it as is for now.